controllers/limits.go

```diff
@@ -0,0 +1,58 @@
+package controller
+
+import (
+	"net/http"
+
+	"github.com/gravitl/netmaker/database"
+	"github.com/gravitl/netmaker/logic"
+	"github.com/gravitl/netmaker/models"
+)
+
+// limit consts
+const (
+	node_l     = 0
+	networks_l = 1
+	users_l    = 2
+	clients_l  = 3
+)
+
+func checkFreeTierLimits(limit_choice int, next http.Handler) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		var errorResponse = models.ErrorResponse{
+			Code: http.StatusUnauthorized, Message: "free tier limits exceeded on networks",
+		}
+
+		if logic.Free_Tier && logic.Is_EE { // check that free tier limits not exceeded
+			if limit_choice == networks_l {
+				currentNetworks, err := logic.GetNetworks()
+				if (err != nil && !database.IsEmptyRecord(err)) || len(currentNetworks) >= logic.Networks_Limit {
+					logic.ReturnErrorResponse(w, r, errorResponse)
```

Comment:
shouldn't the limit check should be `>` rather than `>=`

Reply:
no because that would imply they were able to make more than the limit previously

Reply:
oh got it my bad that is a middleware func which checks before creating resources, makes sense